src/components/Property/Types/VectorProperty/ViewOptions/ColorView.tsx

```diff
@@ -20,38 +39,71 @@ export function ColorView({
   additionalData,
   isInt
 }: Props) {
-  if ((value.length !== 3 && value.length !== 4) || isInt) {
-    throw Error('Invalid use of Color view option!');
-  }
+  const [isError, setIsError] = useState(false);
+
+  const isOutsideRange = value.some((v) => v < 0 || v > 1);
+  const currentColor: RGBA = openspaceColorToRgba(value);
   const hasAlpha = value.length === 4;
+  const valueDisplayString = value
+    .map((v) => parseFloat(v.toFixed(3)).toString())
+    .join(', ');
+
+  validateInput(value, additionalData, isInt);
+
+  function isValidColorString(colorString: string): boolean {
+    const newValues = colorString.split(',').map((v) => parseFloat(v));
+    const result =
+      newValues.length === value.length &&
+      newValues.every((v) => !isNaN(v) && isNumber(v));
+
+    return result;
+  }
+
+  function onColorPickerChange(color: RGBA) {
+    const newValue = toOpenspaceColor(color, hasAlpha);
+    setPropertyValue(newValue);
+
+    setIsError(!isValidColorString(newValue.join(', ')));
+  }
+
+  function handleEnter(newValue: string) {
+    const newNumberValue = newValue.split(',').map((v) => parseFloat(v));
+    setPropertyValue(newNumberValue);
+  }
+
+  function handleInputChange(newValue: string) {
+    const isValid = isValidColorString(newValue);
+    setIsError(!isValid);
+  }
+
+  function handleBlur() {
+    setIsError(!isValidColorString(value.join(', ')));
+  }
 
   return (
-    <Flex gap={'xs'} align={'center'}>
-      <VectorDefaultView
-        disabled={readOnly}
-        setPropertyValue={setPropertyValue}
-        value={value}
-        additionalData={additionalData}
-      />
-      <ColorPicker
-        disabled={readOnly}
-        withAlpha={hasAlpha}
-        color={{
-          r: Math.round(255 * value[0]),
-          g: Math.round(255 * value[1]),
-          b: Math.round(255 * value[2]),
-          a: hasAlpha ? value[3] : 1.0
-        }}
-        onChange={(rgbaColor: RGBA) => {
-          const newValue = [rgbaColor.r / 255, rgbaColor.g / 255, rgbaColor.b / 255];
-          if (hasAlpha) {
-            newValue.push(rgbaColor.a);
-          }
-          // @TODO (emmbr26, 2025-03-14) Check for min max values. color values should be
-          // between 0 and 1.
-          setPropertyValue(newValue);
-        }}
-      />
-    </Flex>
+    <StringInput
+      onEnter={handleEnter}
+      onInput={(e) => handleInputChange(e.currentTarget.value)}
+      value={valueDisplayString}
+      error={isError}
+      leftSection={
+        <ColorPicker
+          color={currentColor}
+          withAlpha={hasAlpha}
+          onChange={onColorPickerChange}
+        />
+      }
+      onBlur={handleBlur}
+      readOnly={readOnly}
+      rightSection={
+        isOutsideRange && (
+          <WarningIcon
+            tooltipText={`Value outside range [${additionalData.min}, ${additionalData.max}]`}
+          />
+        )
+      }
+      placeholder={'Example: 1, 0.5, 0.2'}
```

Comment:
How about inclusing in the placeholder whether the input string should include alpha or not? 
```suggestion
      placeholder={`RGB${hasAlpha ? 'A' : ''}: 1, 0.5, 0.2${hasAlpha ? ', 0.5' : ''}`}
```
![image](https://github.com/user-attachments/assets/31bc9729-6926-472d-96a7-b4d7b7470c1e)
![image](https://github.com/user-attachments/assets/bcf6f53f-3645-4793-b8e0-3ef43fa9c1af)

Reply:
Maybe this is a bit confusing? You might think you need to write "RGBA". But i agree that the placeholder should have the fourth number

Reply:
I get your point, and maybe it's obvious to the user what each value is